diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9660f55..88a7dac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,8 +35,6 @@ ELSE()
   ELSEIF (CMAKE_SYSTEM_PROCESSOR MATCHES 
           "(arm)|(ARM)|(armhf)|(ARMHF)|(armel)|(ARMEL)")
     add_definitions (-march=armv7-a)
-  ELSE ()
-    add_definitions (-march=native) #TODO use correct c++11 def once everybody has moved to gcc 4.7 # for now I even removed std=gnu++0x
   ENDIF()
   add_definitions (
     -O3
diff --git a/src/absolute_pose/modules/Epnp.cpp b/src/absolute_pose/modules/Epnp.cpp
index d52e1ec..4e5e30d 100644
--- a/src/absolute_pose/modules/Epnp.cpp
+++ b/src/absolute_pose/modules/Epnp.cpp
@@ -751,7 +751,6 @@ opengv::absolute_pose::modules::Epnp::qr_solve(
     if (eta == 0)
     {
       A1[k] = A2[k] = 0.0;
-      cerr << "God damnit, A is singular, this shouldn't happen." << endl;
       return;
     }
     else
